source/OptimaJet.DWKit.StarterApplication/Services/SparkPostSender.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentEmail.Core;
+using FluentEmail.Core.Interfaces;
+using FluentEmail.Core.Models;
+using SparkPostDotNet;
+using SparkPostDotNet.Transmissions;
+
+namespace OptimaJet.DWKit.StarterApplication.Services
+{
+    public class SparkPostSender : ISender
+    {
+        public SparkPostClient PostClient { get; }
+        public SparkPostSender(SparkPostClient postClient)
+        {
+            PostClient = postClient;
+        }
+
+
+        public SendResponse Send(IFluentEmail email, CancellationToken? token = null)
+        {
+            return SendAsync(email, token).GetAwaiter().GetResult();
+        }
+
+        public async Task<SendResponse> SendAsync(IFluentEmail email, CancellationToken? token = null)
+        {
+            SendResponse response = new SendResponse();
+            try
+            {
+                var tx = CreateTransmission(email);
+                await PostClient.CreateTransmission(tx);
+            }
+            catch (Exception ex)
+            {
+                response.ErrorMessages.Add(ex.Message);
+            }
+
+            return response;
+        }
+
+        private Transmission CreateTransmission(IFluentEmail email)
+        {
+            if (email.Data.ToAddresses.Count == 0) throw new Exception("Email must have at least 1 To address");
```

Comment:
I have had occasions where I have sent all BCC recipients.  Might want to change the check to be must have at least 1 To, CC or BCC recipient